aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -562,14 +571,16 @@ fn native_string_concat(
     if prefix
         .len()
         .checked_add(suffix.len())
-        .map_or(false, |v| v > CONCAT_PREFIX_AND_SUFFIX_MAX_LENGTH)
+        .map_or(false, |v| v > STRING_SNAPSHOT_INPUT_MAX_LENGTH)
     {
         return Err(SafeNativeError::Abort {
-            abort_code: ECONCAT_STRING_LENGTH_TOO_LARGE,
+            abort_code: EINPUT_STRING_LENGTH_TOO_LARGE,
         });
     }
 
-    context.charge(STRING_UTILS_PER_BYTE * NumBytes::new((prefix.len() + suffix.len()) as u64))?;
+    context.charge(
```

Comment:
I think we can charge twice? Once after we got the prefix, and once after the suffix? Otherwise we do not charge for `string_to_bytes` which ended up being too large.

Reply:
and if we could also decide to move charging before return-s in the > limit case, the same way as above.